### ISSUE TITLE
Redis Interface

### ISSUE THREAD

**Issue by gmonk63** [tidwall/tile38#290]
I didn't want to hijack the previous topic. So i decided to ask directly.  Another user mentioned solely using the the redis interface  and not using any webhooks.  Is this possible with Geofences?  I want to be able to get the entry,exit  notifications without the need for hooks if possible. In my situation the Raspberry 3.14  is the device running Tile and should only worry about up coming Geofence locations and act based on entry/exit of each.  Right now my code grabs the gps coords and updates Tile and when an entry is detected sends a mqtt webhook to a local broker on the  3.14  then makes a decision thereafter.  But It would be nice to be able to sub to the notifications without the need for the broker. I know Redis has the PUBSUB feature but I did not see this listed in the commands for Tile. For exampe lf I setup Fences LOC1 - LOC500   and  sub LOC*   to get all notifications as each Geofence is entered/exited that would be great .

Thanks and sorry for the long post

**Comment by tidwall** [comment]
Did you end up figuring it out?

**Comment by gmonk63** [comment]
No I didnt. Wasnt sure this was possible with tile since it did not look like it supported such features as pub/sub.

**Comment by tidwall** [comment]
There no pubsub in tile38. The way to connect to a geofence over tcp is to send a regular query with the FENCE command. You will get an response indicating success. Then the server changes the protocol to json-lines. So each geofence notification will then come over as a json block on its own line.

**Comment by sbuettner** [comment]
@tidwall Is there support for multiplexing of multiple fences via a single connection to tile38. It looks like a single connection is currently limited to a single fence.

**Comment by sbuettner** [comment]
@tidwall Is there support for multiplexing of multiple fences via a single connection to tile38. It looks like a single connection is currently limited to a single fence.

**Comment by tidwall** [comment]
@sbuettner It's possible to have multiple geofences to a share the same endpoint.

```
SETHOOK hook1 http://10.0.20.78/endpoint NEARBY fleet ...
SETHOOK hook2 http://10.0.20.78/endpoint INTERSECTS busline ...
```

Then `http://10.0.20.78/endpoint` will receive notifications for hook1 and hook2.

**Comment by sbuettner** [comment]
@tidwall Thanks for the response. I was thinking more about a case where a single client receives data for multiple geofences using a single connection without having the server to know about the client IP address or host.

**Comment by sbuettner** [comment]
@tidwall I though about a client-server scenario where the client might proxy the results of the geofences back to its clients like a proxy. It would be nice to if a single connection could be used to query the server and receive the results of a geofence via a single redis-like connection. The server could return for example all objects for a bounding box search and then continously stream the results of the geofence for all objects in that bounding box.

**Comment by tidwall** [comment]
It’s certainly a different approach. It’s more like a Redis style pubsub. Which may be a better way of doing it.

**Comment by sbuettner** [comment]
Exactly, basically a subscription for a geofence: https://redis.io/commands/psubscribe

**Comment by tobilg** [comment]
Hm, I thought using the same endpoint in each hook could achieve something similar @sbuettner

**Comment by tidwall** [comment]
@tobilg You’re right that for webhooks geofences you can point many hooks to one endpoint. But simple client-server geofences are limited to one connection per geofence.

**Comment by gmonk63** [comment]
Is this going to be added ?  I think this would make things so much easier to be able to sub to multiple fence notifications.  I would actually put money towards this being added :)

**Comment by tidwall** [comment]
Yes. I absolutely want to add this feature. 

>  I would actually put money towards this being added :)

That is certainly a good way expedite. :) You're already a backer of the project, but the [Patreon campaign](https://patreon.com/tidwall) does have higher tiers. *wink wink*

Also don't forget to join the [Slack channel](http://tile38.com/slack)! I'm hoping that it's a good alternative to Github Issues for general questions and support.

**Comment by gmonk** [comment]
@tidwall  does 1.13 address this ?  So are we now able to sub to the webhook via the redis client

**Comment by tidwall** [comment]
@gmonk63 Yes. You can use Redis-style pub/sub.

**Comment by gmonk** [comment]
Whoo-hooo   you just made it on my Christmas list... Thanks for all your hard work

**Comment by tidwall** [comment]
You're welcome and I hope it works out for your use-case.
